worker/src/globals/tokio_handle.rs

```diff
@@ -0,0 +1,97 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+    Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use lazy_static::lazy_static;
+use parking_lot::RwLock;
+use tokio::runtime::Handle;
+
+lazy_static! {
+    static ref TOKIO_HANDLE: RwLock<Option<tokio::runtime::Runtime>> = RwLock::new(None);
+}
+
+/// Wrapper for accessing a tokio handle
+pub trait GetTokioHandle {
+    fn get_handle(&self) -> Handle;
+}
+
+/// implementation, using a static global variable internally
+///
+pub struct GlobalTokioHandle;
+
+/// these are the static (global) accessors
+/// reduce their usage where possible and use an instance of TokioHandleAccessorImpl or the trait
+impl GlobalTokioHandle {
+    /// this needs to be called once at application startup!
+    pub fn initialize() {
+        let rt = tokio::runtime::Runtime::new().unwrap();
+        *TOKIO_HANDLE.write() = Some(rt);
+    }
+
+    /// static / global getter of the handle (try to keep private!, use trait to access handle)
+    fn read_handle() -> Handle {
+        TOKIO_HANDLE
+            .read()
+            .as_ref()
+            .expect("Tokio handle has not been initialized!")
+            .handle()
+            .clone()
+    }
+}
+
+impl GetTokioHandle for GlobalTokioHandle {
+    fn get_handle(&self) -> Handle {
+        GlobalTokioHandle::read_handle()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    #[test]
+    fn given_initialized_tokio_handle_when_runtime_goes_out_of_scope_then_blocking_handle_is_valid()
+    {
+        {
+            GlobalTokioHandle::initialize();
+        }
+
+        let handle = GlobalTokioHandle {}.get_handle();
```

Comment:
Do you really like that ` {}` in the notation? I think it disrupts the reader's flow.

Reply:
No you're right - I'm still unsure in which cases I can omit the `{}`. But here it's possible apparently. I'll push that change as soon as GA is finished running, to not start over again and lose the progress.

Reply:
You can push already, it will trigger a new additional run